pontoon/actionlog/models.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import absolute_import, unicode_literals
+
+from django.core.exceptions import ValidationError
+from django.db import models
+
+from pontoon.base.models import (
+    Entity,
+    Locale,
+    Translation,
+    User,
+)
+
+
+class ActionLog(models.Model):
+    ACTIONS_TYPES = (
+        # A translation has been created.
+        ('translation:created', 'Translation created'),
+
+        # A translation has been deleted.
+        ('translation:deleted', 'Translation deleted'),
+
+        # A translation has been approved.
+        ('translation:approved', 'Translation approved'),
+
+        # A translation has been unapproved.
+        ('translation:unapproved', 'Translation unapproved'),
+
+        # A translation has been rejected.
+        ('translation:rejected', 'Translation rejected'),
+
+        # A translation has been unrejected.
+        ('translation:unrejected', 'Translation unrejected'),
+    )
+
+    action_type = models.CharField(max_length=50, choices=ACTIONS_TYPES)
+    created_at = models.DateTimeField(auto_now_add=True)
+    performed_by = models.ForeignKey(User, models.CASCADE)
+
+    # Used to track on what translation related actions apply.
+    translation = models.ForeignKey(
+        Translation,
+        models.CASCADE,
+        blank=True,
+        null=True,
+    )
+
+    # Used when a translation has been deleted.
+    entity = models.ForeignKey(
+        Entity,
+        models.CASCADE,
+        blank=True,
+        null=True,
+    )
+    locale = models.ForeignKey(
+        Locale,
+        models.CASCADE,
+        blank=True,
+        null=True,
+    )
```

Comment:
Should we use this opportunity to introduce the `deleted` flag on the `Translation` model instead of actually deleting entries from the DB? Eventually, that should also allow us to drop `ChangedEntityLocale` model.

Reply:
Do we have a bug filed for that? I can't find one.

I think it's a good idea but it's going to have some impact on our code. We'll need to change a bunch of queries to exclude `deleted=true`.

If `ChangedEntityLocale` is only used to check what locales / entities have changed in a given period of time, then the `ActionLog` table as it is right now is enough to replace it, I believe.

Reply:
We don't have a bug on file for that.

If you agree it's a good idea, I can file it. Fixing it would simplify this patch, but it's not a trivial fix, so we can also do it in a followup.

Reply:
Filed https://bugzilla.mozilla.org/show_bug.cgi?id=1599377